### ISSUE TITLE
Rule Change: `func-style`: add suggestions

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/eslint#15392]
### What rule do you want to change?

func-style

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint func-style: ["error", "expression"]*/

function foo() {
    // ...
}
```

### What does the rule currently do for this code?

Lint error with no suggestions.

### What will the rule do after it's changed?

Lint error with suggestion to convert to a function expression.

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
